app/controllers/text_components_controller.rb

```diff
@@ -86,11 +90,19 @@ def set_index_data
       @text_components = TextComponent.includes(:triggers, :question_answers, :channels)
 
       filter_text_components
+      
+      @trigger_groups = @text_components.order('from_day').group_by { |t| t.trigger_ids }
+      @trigger_groups = @trigger_groups.map { |trigger_ids, components| [ Trigger.find(trigger_ids), components ] }.to_h
+      @trigger_groups = @trigger_groups.sort do |a,b|
+        if(a.first.first && b.first.first)
+          a.first.first.name.downcase <=> b.first.first.name.downcase
```

Comment:
Using `sort_by` is not possible as text components without a trigger need to be sorted to the end. However, I have cleaned up the code for the controller and the template a bit and followed your other tips. Hope this is now easier to read and understand.